test/IRGen/objc_super.swift

```diff
@@ -136,5 +136,5 @@ class GenericRuncer<T> : Gizmo {
 }
 
 // CHECK: define internal swiftcc void [[PARTIAL_FORWARDING_THUNK]](%swift.refcounted* swiftself %0) {{.*}} {
-// CHECK: @"$ss12StaticStringV14withUTF8BufferyxxSRys5UInt8VGXElFyAEcfU_"
+// CHECK: @"$ss12StaticStringV14withUTF8BufferyxxSRys5UInt8VGXElFxAFXEfU_"
```

Comment:
This test now has a different `closure #1` type in its mangled name.

* Old swift-demangle result:

  ```
  $ss12StaticStringV14withUTF8BufferyxxSRys5UInt8VGXElFyAEcfU_ ---> 
  closure #1 (Swift.UInt8) -> () in 
  Swift.StaticString.withUTF8Buffer<A>((Swift.UnsafeBufferPointer<Swift.UInt8>) -> A) -> A
  ```

* New swift-demangle result:

  ```
  $ss12StaticStringV14withUTF8BufferyxxSRys5UInt8VGXElFxAFXEfU_ ---> 
  closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> A in 
  Swift.StaticString.withUTF8Buffer<A>((Swift.UnsafeBufferPointer<Swift.UInt8>) -> A) -> A
  ```